pkg/image/registry/imagesignature/strategy.go

```diff
@@ -0,0 +1,74 @@
+package imagesignature
+
+import (
+	"crypto/sha1"
+	"fmt"
+
+	"github.com/golang/glog"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	"github.com/openshift/origin/pkg/image/api/validation"
+)
+
+// strategy implements behavior for ImageStreamTags.
+type strategy struct {
+	runtime.ObjectTyper
+}
+
+var Strategy = &strategy{
+	ObjectTyper: kapi.Scheme,
+}
+
+func (s *strategy) NamespaceScoped() bool {
+	return false
+}
+
+func (s *strategy) PrepareForCreate(obj runtime.Object) {
+	signature := obj.(*imageapi.ImageSignature)
+
+	signature.Conditions = nil
+	signature.ImageIdentity = ""
+	signature.SignedClaims = nil
+	signature.Created = nil
+	signature.IssuedBy = nil
+	signature.IssuedTo = nil
+
+	// generate signature name if it contains just image name
+	imageName, sigName, _ := imageapi.SplitImageSignatureName(signature.Name)
+	glog.Infof("imageName=%q, sigName=%q, signature.Name=%q", imageName, sigName, signature.Name)
+	if len(imageName) == 0 || len(sigName) > 0 {
+		return
+	}
+	sigName = fmt.Sprintf("%x", sha1.Sum([]byte(fmt.Sprintf("%s:%s", signature.Type, signature.Content))))
```

Comment:
I'm not clear on why we're stomping the name with this.  I thought we decided that users had to choose names for these things so they'd later be able to delete them.  This makes the name unpredictable.